Assets/Scripts/Enemy/WanderState.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class WanderState : State
+{
+    public GameObject[] waypoints;
+
+    public GameObject Lantern;
+
+    public bool chasing;
+
+    public float range;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        FindNextPoint();
+        AddTransition<ChaseState>(() => chasing);
+    }
+
+    // Update is called once per frame
+    public override void Update()
+    {
+        if (Vector3.Distance(transform.position, controller.destPos) <= 3.0f)
+        {
+            FindNextPoint();
+        }
+    }
+
+    public void FixedUpdate()
+    {
+        Debug.DrawRay(Lantern.transform.position, Lantern.transform.forward * range, Color.black);
+        if (Physics.Raycast(Lantern.transform.position, Lantern.transform.forward * range, out RaycastHit hit))
+        {
+            if (hit.collider.CompareTag("Player"))
+            {
+                chasing = true;
+            }
+        }
+    }
+
+    public override void OnDisable()
+    {
+        base.OnDisable();
+        chasing = false;
+    }
+
+    public void FindNextPoint()
+    {
+        int rndIndex = Random.Range(0, waypoints.Length);
+        Vector3 rndPosition = Vector3.zero;
+        controller.destPos = waypoints[rndIndex].transform.position + rndPosition;
```

Comment:
```suggestion
        controller.destPos = waypoints[rndIndex].transform.position + rndPosition;
```
`rndPosition` is 0,0,0, dus je voegt 0 toe aan alle coördinaten

Reply:
die 0,0,0 is er om wanneer er geen destpos gezet is dat ie naar 0,0,0 gaat